### ISSUE TITLE
ClearSpeak bugs

### ISSUE THREAD

**Issue by NSoiffer** [zorkow/speech-rule-engine#284]
## Description
The follow are bugs in the speech output for ClearSpeak.

MathML: `msup><mi extid="2jbc">x</mi><mi extid="2jbd">n</mi></msup>`
"x to the n t h power" -- highlighting is ok
SRE Speech: `<?xml version="1.0"?><speak version="1.1" xmlns="http://www.w3.org/2001/10/synthesis"><prosody rate="100%"> <mark name="2jbc"/> <say-as interpret-as="character">x</say-as> to the <mark name="2jbd"/> nth power <break time="250ms"/> </prosody></speak>`
I think I had to do some tricks in MathPlayer to get this to sound right (e.g., might have used "enth", etc.).

MathML: `mrow><mn extid="i10k5n">2</mn><mo>&#x2064;</mo><mfrac><mn extid="i10k5o">3</mn><mi extid="i10k5p">p</mi></mfrac></mrow>`
incorrect interpretation as mixed fractions.
SRE Speech: `<?xml version="1.0"?><speak version="1.1" xmlns="http://www.w3.org/2001/10/synthesis"><prosody rate="100%"> <mark name="i10k5n"/> 2 plus <break time="250ms"/> <mark name="i10k5o"/> 3 over <mark name="i10k5p"/> <say-as interpret-as="character">p</say-as> <break time="250ms"/> </prosody></speak>`
Also, in English, mixed fractions would typically be spoken as "2 and 2/3" not "2 plus 2/3" (which is what the ClearSpeak spec says to say).

MathML: `mrow><mi extid="hm27tf">x</mi><mo extid="hm27tg">∈</mo><msup><mi mathvariant="double-struck" extid="hm27th">R</mi><mn extid="hm27ti">2</mn></msup></mrow>`
Should say "r two" for R^2
SRE Speech: `<?xml version="1.0"?><speak version="1.1" xmlns="http://www.w3.org/2001/10/synthesis"><prosody rate="100%"> <mark name="hm27tf"/> <say-as interpret-as="character">x</say-as> <mark name="hm27tg"/> is a member of <mark name="hm27th"/> the real numbers squared </prosody></speak>`

MathML: `mrow><mi extid="hqgkk4">x</mi><mo extid="hqgkk5">∈</mo><msup><mi mathvariant="double-struck" extid="hqgkk6">Z</mi><mi extid="hqgkk7">+</mi></msup></mrow>`
Should say "Positive integers"
SRE Speech: `<?xml version="1.0"?><speak version="1.1" xmlns="http://www.w3.org/2001/10/synthesis"><prosody rate="100%"> <mark name="hqgkk4"/> <say-as interpret-as="character">x</say-as> <mark name="hqgkk5"/> is a member of <mark name="hqgkk6"/> the integers raised to the <mark name="hqgkk7"/> plus power </prosody></speak>`

MathML: `mrow><msup><mi extid="byvddo">x</mi><mn extid="byvddp">2</mn></msup><mo extid="byvddq">+</mo><mn extid="byvddr">2</mn><mo>&InvisibleTimes;</mo><mi extid="byvdds">x</mi><msub><mo extid="byvddt">∣</mo><mrow><mi extid="byvddu">x</mi><mo extid="byvddv">=</mo><mn extid="byvddw">3</mn></mrow></msub></mrow>`
Should say "evaluated at"
SRE Speech: `<?xml version="1.0"?><speak version="1.1" xmlns="http://www.w3.org/2001/10/synthesis"><prosody rate="100%"> <mark name="byvddo"/> <say-as interpret-as="character">x</say-as> squared <mark name="byvddq"/> plus <mark name="byvddr"/> 2 <mark name="byvdds"/> <say-as interpret-as="character">x</say-as> <break time="250ms"/> <mark name="byvddt"/> bar sub <mark name="byvddu"/> <say-as interpret-as="character">x</say-as> <mark name="byvddv"/> equals <mark name="byvddw"/> 3 <break time="250ms"/> </prosody></speak>`

MathML: `mrow><msup><mi extid="e842v2">x</mi><mn extid="e842v3">2</mn></msup><mo extid="e842v4">+</mo><mn extid="e842v5">2</mn><mo>&InvisibleTimes;</mo><mi extid="e842v6">x</mi><mo>&InvisibleTimes;</mo><msub><mo extid="e842v7">∣</mo><mrow><mi extid="e842v8">x</mi><mo extid="e842v9">=</mo><mn extid="e842va">3</mn></mrow></msub></mrow>`
This variant of the above didn't speak the subscript:
SRE Speech: `<?xml version="1.0"?><speak version="1.1" xmlns="http://www.w3.org/2001/10/synthesis"><prosody rate="100%"> <mark name="1q4q0k"/> <say-as interpret-as="character">x</say-as> <prosody pitch="17%"> squared </prosody> <break time="300ms"/> <mark name="1q4q0m"/> plus <mark name="1q4q0n"/> two times <mark name="1q4q0o"/> <say-as interpret-as="character">x</say-as> postfix bar </prosody></speak>`

**Comment by zorkow** [comment]
Thanks for those. Most should be easy to fix.

I am reluctant to tinker with `nth`. Firstly, pronunciation differs from TTS. It's a bit like the `[ae]` vs `[e\i]` problem, e.g, in `2a`. Secondly, I know people that use SRE to generate transcriptions that are never meant to be voiced (at least not by a TTS).  I could see this becoming part of the finalization in a renderer class, but it should definitely stay out of the actual rules.

For the mixed number: So you would actually put an invisible plus there? I've never considered that case. Currently an invisible plus is just a plus, hence the result.

**Comment by zorkow** [comment]
That dropped off the radar, but I finally got round to fixing the issues. See PR #408.

I am thinking about fixing the pronunciation issue in an overhaul of the `SSML` driver using the `sub` tag and alias in a future PR.

The evaluates problem is actually down to incorrect use of  Unicode. U+2223 is really divides and U+007C should be used for the evaluation bar. But I've added it to be considered as possible bar now as well.
